lib/db.js

```diff
@@ -36,7 +37,40 @@ const actionOneOrMany = curry(async (action, collection, isMultiple, ...params)
 });
 const insertOneOrMany = actionOneOrMany('insert');
 const updateOneOrMany = actionOneOrMany('update');
-const removeOneOrMany = actionOneOrMany('remove');
+const deleteOneOrMany = actionOneOrMany('delete');
+
+const deserialize = (data = {}) => {
+  if (Array.isArray(data)) {
+    return data.map(deserialize);
+  }
+  if (data === null) { return; }
+  if (typeof data !== 'object') {
+    return data;
+  }
+
+  if (data.__type === 'bson.ObjectID') {
+    return new ObjectId(data.id);
+  }
+
+  if (data.__type === 'Date') {
+    return new Date(data.valueOf());
+  }
+
+  const keys = Object.keys(data);
+  if (keys.length === 1) {
+    if (keys[0] === '$oid') {
+      return new ObjectId(data[keys[0]]);
+    }
+
+    if (keys[0] === '$date') {
```

Comment:
do we actually need to do $date? I'd almost rather fix the places where we are handing in $date